### PR TITLE
Vaadin security enhancements

### PR DESCRIPTION
**Pull request by bogdan-bondar**
* removed VaadinManagedSecurity configuration from MgmtUiAutoConfiguration
* added AccessDecisionManager to UI security configuration in order to support method security in UI in context of VaadinSharedSecurity
* Changed UI push transport from WEBSOCKET to WEBSOCKET_XHR to solve problems with Spring Security Context
* Suppressed atmosphere IOUtils false-positive warning
* Removed obsolete AsyncVaadinServletConfiguration
* Defined Vaadin4SpringServlet bean instead of plain SpringVaadinServlet for configuration flexibility
* Removed obsolete SpringSecurityAtmosphereInterceptor because the client does not communicate with the server using websocket protocol anymore
* Removed unit test for SpringSecurityAtmosphereInterceptor

Signed-off-by: Bogdan Bondar <Bogdan.Bondar@bosch.io>

### CODE REVIEW

**Comment by stefbehl** [comment]
@hawkbit-bot verify please

File: hawkbit-runtime/hawkbit-update-server/src/main/java/org/eclipse/hawkbit/app/MyUI.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
  *
  */
 @SpringUI
-@Push(value = PushMode.AUTOMATIC, transport = Transport.WEBSOCKET)
+@Push(value = PushMode.AUTOMATIC, transport = Transport.WEBSOCKET_XHR)
```
Comment (stefbehl) [inline hawkbit-runtime/hawkbit-update-server/src/main/java/org/eclipse/hawkbit/app/MyUI.java:39]: A comment explaining why we use "WEBSOCKET_XHR" (as opposed to plain "WEBSOCKET") could be helpful

**Review by stefbehl — APPROVED** [review]
Looks good - thanks Bogdan!
Let's wait for the hawkbit-bot to finish his analysis.

File: hawkbit-runtime/hawkbit-update-server/src/main/java/org/eclipse/hawkbit/app/MyUI.java (hunk 1 above)
Comment (bogdan-bondar) [reply hawkbit-runtime/hawkbit-update-server/src/main/java/org/eclipse/hawkbit/app/MyUI.java:39]: done

**Pull request merged**